mmf/utils/checkpoint.py

```diff
@@ -194,7 +194,25 @@ def _load(self, file, force=False, load_zoo=False, load_pretrained=False):
         state_dict = self.upgrade_state_dict(ckpt["model"])
 
         if len(pretrained_state_mapping.items()) == 0:
-            self.trainer.model.load_state_dict(state_dict, strict=False)
+            incompatible_keys = self.trainer.model.load_state_dict(
+                state_dict, strict=False
+            )
+            assert len(incompatible_keys.missing_keys) == 0, (
```

Comment:
Can we show a warning for this instead of assert? Sometimes some extra params get added to models. Recently I discovered with latest version of transformers that they have added a new variable to store position ids. That will make our old checkpoints throw error.

Reply:
Hmm, I do think this is good assert validation, but yeah, warning sounds good.